src/sync/SplitUpdateWorker/index.js

```diff
@@ -57,8 +62,12 @@ export default class SplitUpdateWorker {
    * @param {string} defaultTreatment default treatment value
    */
   killSplit(changeNumber, splitName, defaultTreatment) {
-    this.splitStorage.killLocally(splitName, defaultTreatment, changeNumber);
-    this.put(changeNumber);
+    this.splitStorage.killLocally(splitName, defaultTreatment, changeNumber).then((updated) => {
+      if (updated) {
+        this.splitsEventEmitter.emit(this.splitsEventEmitter.SDK_SPLITS_KILL);
```

Comment:
After reading my comments on readiness read this:

Here you can emit the SDK_SPLITS_ARRIVED (since actually you've received splits data) but passing the boolean flag as true for readiness to understand it doesn't need to change the status of Splits. 

No need to pass that boolean through the synchronizeSplits + updater path either after that change unless I'm missing something, but seems achievable.

Reply:
done